src/System.IO.Compression.Brotli/System/IO/Compression/BrotliStream.cs

```diff
@@ -138,26 +154,21 @@ public override long Position
 
         protected virtual void FlushEncoder(bool finished)
         {
-            if (_encoder.State == IntPtr.Zero) return;
-            if (BrotliNative.BrotliEncoderIsFinished(_encoder.State)) return;
-            BrotliEncoderOperation op = finished ? BrotliEncoderOperation.Finish : BrotliEncoderOperation.Flush;
-            nuint totalOut = 0;
-            while (true)
+            if (_state.BrotliNativeState == IntPtr.Zero) return;
+            if (BrotliNative.BrotliEncoderIsFinished(_state.BrotliNativeState)) return;
+            _nextInput = new byte[0];
```

Comment:
Why allocate an empty array and throw away the one you allocated in the constructor?